### ISSUE TITLE
Version 1.53j, ij.io.Opener no longer works on headless

### ISSUE THREAD

**Issue by rudolphi** [imagej/imagej1#140]
Throws
```
Caused by: java.awt.HeadlessException
	at java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:204)
	at java.awt.MenuComponent.<init>(MenuComponent.java:163)
	at java.awt.MenuItem.<init>(MenuItem.java:214)
	at java.awt.MenuItem.<init>(MenuItem.java:198)
	at java.awt.Menu.<init>(Menu.java:157)
	at java.awt.Menu.<init>(Menu.java:137)
	at ij.Menus.getMenu(Menus.java:849)
	at ij.Menus.getMenu(Menus.java:839)
	at ij.Menus.addMenuBar(Menus.java:109)
	at ij.IJ.init(IJ.java:422)
	at ij.Menus.getCommands(Menus.java:1339)
	at ij.io.Opener.<clinit>(Opener.java:47)
	...
```

**Comment by ctrueden** [comment]
@rudolphi OK, I transferred the issue back to the original ImageJ repository, then.

With which version of ImageJ are you able to use `ij.io.Opener` headless?

**Comment by rudolphi** [comment]
It works smoothly with 1.53i

**Comment by ctrueden** [comment]
@rudolphi Ah I see, I was testing the wrong thing. I was using `Opener#open(String)`, but that tries to show the image. The `Opener#openImage(String)` method, which returns `ImagePlus`, does indeed work headless with 1.53i for me:
```
$ echo 'public class X { public static void main(String[] args) { System.out.println(new ij.io.Opener().openImage(args[0])); } }' > X.java
$ mvn dependency:get -Dartifact=net.imagej:ij:1.53i
$ javac -cp ~/.m2/repository/net/imagej/ij/1.53i/ij-1.53i.jar X.java
$ java -Djava.awt.headless=true -cp ~/.m2/repository/net/imagej/ij/1.53i/ij-1.53i.jar:. X ~/data/blobs.tif
img["blobs.tif" (-2), 8-bit, 256x254x1x1x1]
```

And with 1.53j:
```
$ mvn dependency:get -Dartifact=net.imagej:ij:1.53i
$ java -Djava.awt.headless=true -cp ~/.m2/repository/net/imagej/ij/1.53j/ij-1.53j.jar:. X ~/data/blobs.tif
Exception in thread "main" java.lang.ExceptionInInitializerError
	at X.main(X.java:1)
Caused by: java.awt.HeadlessException
	at java.desktop/java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:165)
	at java.desktop/java.awt.MenuComponent.<init>(MenuComponent.java:181)
	at java.desktop/java.awt.MenuItem.<init>(MenuItem.java:224)
	at java.desktop/java.awt.MenuItem.<init>(MenuItem.java:208)
	at java.desktop/java.awt.Menu.<init>(Menu.java:163)
	at java.desktop/java.awt.Menu.<init>(Menu.java:144)
	at ij.Menus.getMenu(Menus.java:849)
	at ij.Menus.getMenu(Menus.java:839)
	at ij.Menus.addMenuBar(Menus.java:109)
	at ij.IJ.init(IJ.java:422)
	at ij.Menus.getCommands(Menus.java:1339)
	at ij.io.Opener.<clinit>(Opener.java:47)
	... 1 more
```

So @rasband this is an issue introduced in b553ff4eb1fa075a4794af057de172712cf46681, which as @rudolphi points out, adds an `IJ.init()` call to `Menus.getCommands()`. Would it make sense to test for headless mode in `Menus#getMenu(String, boolean)` and return `null` in that case?

**Comment by rasband** [comment]
This bug should be fixed in the ImageJ 1.53o20 daily build. The Menus.getCommands() method now has a test for headless mode.

```
   public static Hashtable getCommands() {
      if (pluginsTable==null && !GraphicsEnvironment.isHeadless())
         IJ.init();
         return pluginsTable;
      }
   }

```
